src/main/java/org/folio/services/item/ItemService.java

```diff
@@ -176,4 +185,27 @@ public ItemService(Context vertxContext, Map<String, String> okapiHeaders) {
   private Function<SQLConnection, Future<RowSet<Row>>> updateSingleItemBatchFactory(Item item) {
     return connection -> itemRepository.update(connection, item.getId(), item);
   }
+
+  private Future<Item> refuseWhenServicePointIdDoesNotExist(final Item item) {
+    if (item.getStatisticalCodeIds().isEmpty()) {
+      return succeededFuture(item);
+    }
+
+    return statisticalCodeRepository.getById(item.getStatisticalCodeIds())
+      .compose(statCodeMap -> {
+        if (statCodeMap.size() == item.getStatisticalCodeIds().size()) {
+          return succeededFuture(item);
+        }
```

Comment:
Does this mean that if the size of the result set is the same as the number of statistical codes in the item record, it is assumed they are all valid?

Reply:
If the number of statistical code ids on the item to create or update matched the number of results of get statistical codes by ids they all exist.